### PR TITLE
Fix update pulling with termination aware locks enabled

### PR DESCRIPTION
**Pull request by lutovich**

### CODE REVIEW

File: enterprise/ha/src/main/java/org/neo4j/kernel/ha/HaRequestType210.java [hunk 1]
```diff
@@ -173,7 +173,26 @@ public boolean isLock()
...
+             2) transaction rolled back successfully
+             3) transaction was terminated
+
+            Master's response for this call is an obligation to pull up to a specified txId.
+            Processing/unpacking of this response is not needed in all 3 cases:
+             1) committed transaction pulls transaction stream as part of COMMIT call
+             2) rolled back transaction does not care about reading any more
+             3) terminated transaction does not care about reading any more
+            */
+            return false;
```
Comment (chrisvest) [inline enterprise/ha/src/main/java/org/neo4j/kernel/ha/HaRequestType210.java:193]: Good comment! 👍

File: enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/SlaveLocksClientTest.java [hunk 2]
```diff
@@ -447,4 +477,11 @@ private SlaveLocksClient stoppedClient()
         client.stop();
         return client;
     }
+
+    private SlaveLocksClient closedClient()
+    {
+        client.close();
+        client.acquireShared( NODE, 1 ); // trigger new lock session initialization
```
Comment (chrisvest) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/SlaveLocksClientTest.java:563]: The acquire does not throw?

**Comment by chrisvest** [comment]
@lutovich I only have one question, otherwise happy with this.

File: enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/SlaveLocksClientTest.java (hunk 2 above)
Comment (lutovich) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/lock/SlaveLocksClientTest.java:563]: Ooh, it does. So tests that I've added use `@Test` with expected exception. And this expected exception was thrown, just not from the place I want. Will swap these two lines and change tests.

**Comment by digitalstain** [comment]
lgtm

**Pull request merged**